jax/_src/cache_key.py

```diff
@@ -129,8 +130,22 @@ def _log_cache_key_hash(hash_obj, last_serialized: str, hashfn):
     )
 
 
+def _remove_custom_partitioning_ptr(m: ir.Module):
```

Comment:
Can you add a comment describing what problem this is solving? (Python function pointers aren't deterministic across executions.)